Pytest cannot load its plugin 'pytest_cov' with latest package level

`make test` fails because pytest cannot load its plugin 'pytest_cov'.

The following traceback is from Travis run: https://travis-ci.org/pywbem/pywbem/jobs/424857470

```
bash -c "set -o pipefail; PYTHONWARNINGS=default py.test --color=yes --cov pywbem --cov pywbem_mock  --cov-config coveragerc  --ignore=attic --ignore=releases -s 2>&1 |tee test_27.tmp.log"
/opt/python/2.7.14/lib/python2.7/pkgutil.py:186: ImportWarning: Not importing directory '/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/sphinxcontrib': missing __init__.py
  file, filename, etc = imp.find_module(subname, path)
Traceback (most recent call last):
  File "/home/travis/virtualenv/python2.7.14/bin/py.test", line 11, in <module>
    sys.exit(main())
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/_pytest/config/__init__.py", line 56, in main
    config = _prepareconfig(args, plugins)
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/_pytest/config/__init__.py", line 181, in _prepareconfig
    pluginmanager=pluginmanager, args=args
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/pluggy/hooks.py", line 258, in __call__
    return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/pluggy/manager.py", line 67, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/pluggy/manager.py", line 61, in <lambda>
    firstresult=hook.spec_opts.get('firstresult'),
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/pluggy/callers.py", line 196, in _multicall
    gen.send(outcome)
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/_pytest/helpconfig.py", line 89, in pytest_cmdline_parse
    config = outcome.get_result()
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/pluggy/callers.py", line 77, in get_result
    _reraise(*ex)  # noqa
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/pluggy/callers.py", line 180, in _multicall
    res = hook_impl.function(*args)
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/_pytest/config/__init__.py", line 617, in pytest_cmdline_parse
    self.parse(args)
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/_pytest/config/__init__.py", line 782, in parse
    self._preparse(args, addopts=addopts)
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/_pytest/config/__init__.py", line 734, in _preparse
    self.pluginmanager.load_setuptools_entrypoints("pytest11")
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/pluggy/manager.py", line 259, in load_setuptools_entrypoints
    message="Plugin %r could not be loaded: %s!" % (ep.name, e))
pluggy.manager.PluginValidationError: Plugin 'pytest_cov' could not be loaded: (coverage 4.0.3 (/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages), Requirement.parse('coverage>=4.4'))!
make: *** [test] Error 1
```

--- a/.gitignore
+++ b/.gitignore
@@ -30,7 +30,7 @@ mofparsetab.py
 moflextab.py
 testsuite/runtests.log
 /testsuite/testtmp/
-/testsuite/schema/mof/
+/testsuite/schema/mof*/
 /epydoc-3.0.1-patches/
 swig-2.0*.tar.gz
 /swig-2.0*/

--- a/.gitignore
+++ b/.gitignore
@@ -40,3 +40,4 @@ setuptools-*.zip
 /tmp_*/
 /dist/pywbem*.tar.gz
 /dist/pywbem*.whl
+.DS_Store

--- a/.travis.yml
+++ b/.travis.yml
@@ -204,7 +204,8 @@ install:
 
 # commands to run builds & tests
 script:
-  - if [[ -n $_MANUAL_CI_RUN ]]; then make build; fi
+# make build is always run in order to verify the package version determination
+  - make build
   - if [[ -n $_MANUAL_CI_RUN ]]; then make builddoc; fi
   - make check
   - if [[ -n $_MANUAL_CI_RUN ]]; then make pylint; fi

--- a/dev-requirements.txt
+++ b/dev-requirements.txt
@@ -22,7 +22,15 @@ decorator>=4.0.11
 yamlordereddictloader>=0.4.0
 
 # Unit test (no imports, invoked via py.test script):
-pytest-cov>=2.4.0
+
+# TODO: Remove the pinning of the pytest-cov version again once issue
+#       https://github.com/z4r/python-coveralls/issues/66
+#       is resolved.
+#       Background: pytest-cov 2.6.0 has increased the version
+#       requirement for the coverage package from >=3.7.1 to
+#       >=4.4, which is in conflict with the version requirement
+#       defined by the python-coveralls package for coverage==4.0.3.
+pytest-cov>=2.4.0,<2.6
 
 # Coverage reporting (no imports, invoked via coveralls script, only used on py27):
 python-coveralls>=2.8.0; python_version == '2.7'

--- a/docs/changes.rst
+++ b/docs/changes.rst
@@ -27,6 +27,10 @@ Released: not yet
 
 **Enhancements:**
 
+* Make: Eliminated the confusing but unproblematic error message about
+  pbr importing when running certain make targets in a freshly created
+  Python environment. Issue #1288.
+
 **Known issues:**
 
 * See `list of open issues`_.

--- a/docs/changes.rst
+++ b/docs/changes.rst
@@ -31,6 +31,13 @@ Released: not yet
   pbr importing when running certain make targets in a freshly created
   Python environment. Issue #1288.
 
+* Pinned the version of the pytest-cov package to <2.6 due to the fact that
+  pytest-cov 2.6.0 has increased its version requirement for the coverage
+  package from coverage>=3.7.1 to coverage>=4.4. That is in conflict with
+  the version requirement of python-coveralls for coverage==4.0.3.
+  This is only a workaround; An issue against python-coveralls has been
+  opened: https://github.com/z4r/python-coveralls/issues/66
+
 **Known issues:**
 
 * See `list of open issues`_.

--- a/makefile
+++ b/makefile
@@ -69,8 +69,12 @@ endif
 # Directory for coverage html output. Must be in sync with the one in coveragerc.
 coverage_html_dir := coverage_html
 
-# Package version (full version, including any pre-release suffixes, e.g. "0.1.0-dev1")
-package_version := $(shell $(PYTHON_CMD) -c "from pbr.version import VersionInfo; print(VersionInfo('pywbem').release_string())")
+# Package version (full version, including any pre-release suffixes, e.g. "0.1.0-dev1")#
+# Note: Some make actions (such as clobber) cause the package version to change,
+# e.g. because the pywbem.egg-info directory or the PKG-INFO file are deleted,
+# when a new version tag has been assigned. Therefore, this variable is assigned with
+# "=" so that it is evaluated every time it is used.
+package_version := $(shell $(PYTHON_CMD) -c $$'try:\n from pbr.version import VersionInfo\nexcept ImportError:\n pass\nelse:\n print(VersionInfo("$(package_name)").release_string())\n')
 
 # Python versions
 python_version := $(shell $(PYTHON_CMD) -c "import sys; sys.stdout.write('%s.%s.%s'%sys.version_info[0:3])")
@@ -297,7 +301,7 @@ develop: develop_os.done _install_basic dev-requirements.txt
 	@echo "makefile: Target $@ done."
 
 .PHONY: build
-build: _check_version $(bdist_file) $(sdist_file)
+build: $(bdist_file) $(sdist_file)
 	@echo "makefile: Target $@ done."
 
 .PHONY: builddoc
@@ -437,7 +441,7 @@ MANIFEST.in: makefile
 # regenerate MANIFEST. Otherwise, changes in MANIFEST.in will not be used.
 # Note: Deleting build is a safeguard against picking up partial build products
 # which can lead to incorrect hashbangs in the pywbem scripts in wheel archives.
-$(bdist_file) $(sdist_file): setup.py MANIFEST.in $(dist_dependent_files) $(moftab_files)
+$(bdist_file) $(sdist_file): _check_version setup.py MANIFEST.in $(dist_dependent_files) $(moftab_files)
 	@echo "makefile: Creating the distribution archive files"
 	rm -rf MANIFEST $(package_name).egg-info .eggs build
 	$(PYTHON_CMD) setup.py sdist -d $(dist_dir) bdist_wheel -d $(dist_dir) --universal
@@ -506,4 +510,3 @@ $(doc_conf_dir)/mof_compiler.help.txt: mof_compiler $(package_name)/mof_compiler
 	@echo "makefile: Creating mof_compiler script help message file"
 	./mof_compiler --help >$@
 	@echo "makefile: Done creating mof_compiler script help message file: $@"
-

--- a/makefile
+++ b/makefile
@@ -74,7 +74,7 @@ coverage_html_dir := coverage_html
 # e.g. because the pywbem.egg-info directory or the PKG-INFO file are deleted,
 # when a new version tag has been assigned. Therefore, this variable is assigned with
 # "=" so that it is evaluated every time it is used.
-package_version := $(shell $(PYTHON_CMD) -c $$'try:\n from pbr.version import VersionInfo\nexcept ImportError:\n pass\nelse:\n print(VersionInfo("$(package_name)").release_string())\n')
+package_version = $(shell $(PYTHON_CMD) -c "$$(printf 'try:\n from pbr.version import VersionInfo\nexcept ImportError:\n pass\nelse:\n print(VersionInfo(\042$(package_name)\042).release_string())\n')")
 
 # Python versions
 python_version := $(shell $(PYTHON_CMD) -c "import sys; sys.stdout.write('%s.%s.%s'%sys.version_info[0:3])")
@@ -84,10 +84,11 @@ python_mn_version := $(shell $(PYTHON_CMD) -c "import sys; sys.stdout.write('%s%
 dist_dir := dist
 
 # Distribution archives
-bdist_file := $(dist_dir)/$(package_name)-$(package_version)-py2.py3-none-any.whl
-sdist_file := $(dist_dir)/$(package_name)-$(package_version).tar.gz
+# These variables are set with "=" for the same reason as package_version.
+bdist_file = $(dist_dir)/$(package_name)-$(package_version)-py2.py3-none-any.whl
+sdist_file = $(dist_dir)/$(package_name)-$(package_version).tar.gz
 
-dist_files := $(bdist_file) $(sdist_file)
+dist_files = $(bdist_file) $(sdist_file)
 
 # Source files in the packages
 package_py_files := \

